lib/scarpe/spacing.rb

```diff
@@ -0,0 +1,39 @@
+class Scarpe
+  module Spacing
+    DIRECTIONS = [:left, :right, :top, :bottom]
+
+    def style
+      styles = (super if defined?(super)) || {}
+
+      extract_spacing_styles_for(:margin, styles, @margin)
+      extract_spacing_styles_for(:padding, styles, @padding)
+
+      styles
+    end
+
+    def extract_spacing_styles_for(attribute, styles, values)
+      values ||= spacing_values_from_options(attribute)
+
+      case values
+      when Hash
+        values.each do |direction, value|
+          styles["#{attribute}-#{direction}"] = Dimensions.length(value)
+        end
+      when Array
+        DIRECTIONS.zip(values).to_h.each do |direction, value|
+          styles["#{attribute}-#{direction}"] = Dimensions.length(value)
+        end
```

Comment:
Nit: What happens here if somebody sends in an array of three elements? Should we compact after the `to_h`?

Reply:
Then only margin-left, margin-right, and margin-top will be set. We could compact here but the `compact!` handles all cases so probably not needed.

Reply:
I benchmarked it real quick and the difference is fairly minor but a complex enough layout (100,000+ stacks) could see a speed boot (~100-200 ms) from compacting post `to_h` so I'm going to make the change. Good call!